proteuslib/property_models/seawater_prop_pack.py

```diff
@@ -493,6 +493,91 @@ def release_state(self, flags, outlvl=idaeslog.NOTSET):
+                         - True - State variables will be fixed
+                         - False - State variables will remain unfixed, unless already fixed
+            outlvl : idaes logger object that sets output level of solve call (default=idaeslog.NOTSET)
+            solver : solver name string if None is provided the default solver
+                     for IDAES will be used (default = None)
+            optarg : solver options dictionary object (default={})
+
+        Returns:
+            results object from state block solve
+        """
+        # Get logger
+        solve_log = idaeslog.getSolveLogger(self.name, level=outlvl, tag="properties")
+
+        # Initialize at current state values (not user provided)
+        self.initialize(solver=solver, optarg=optarg, outlvl=outlvl)
+
+        # Set solver and options
+        opt = get_solver(solver, optarg)
+
+        # Fix variables and check degrees of freedom
+        flags = {}  # dictionary noting which variables were fixed and their previous state
+        for k in self.keys():
+            sb = self[k]
+            for (v_name, ind), val in var_args.items():
+                var = getattr(sb, v_name)
+                if iscale.get_scaling_factor(var[ind]) is None:
+                    _log.warning(
+                            "While using the calculate_state method on {sb_name}, variable {v_name} "
+                            "was provided as an argument in var_args, but it does not have a scaling "
+                            "factor. This suggests that the calculate_scaling_factor method has not been "
+                            "used or the variable was created on demand after the scaling factors were "
+                            "calculated. It is recommended to touch all relevant variables (i.e. call "
+                            "them or set an initial value) before using the calculate_scaling_factor "
+                            "method.".format(v_name=v_name, sb_name=sb.name))
+                if var[ind].is_fixed():
+                    flags[(k, v_name, ind)] = True
+                    if value(var[ind]) != val:
+                        raise ConfigurationError(
+                            "While using the calculate_state method on {sb_name}, {v_name} was "
+                            "fixed to a value {val}, but it was already fixed to value {val_2}. "
+                            "Unfix the variable before calling the calculate_state "
+                            "method or update var_args."
+                            "".format(sb_name=sb.name, v_name=var.name, val=val, val_2=value(var[ind])))
+                else:
+                    flags[(k, v_name, ind)] = False
+                    var[ind].fix(val)
+
+            if degrees_of_freedom(sb) != 0:
+                raise RuntimeError("While using the calculate_state method on {sb_name}, the degrees "
```

Comment:
There's a new function in `proteuslib.util.initialization` called `check_dof` which can be used to handle these checks now. `check_dof` will check for DOF=0 and you can specify whether you want a warning or an error to show up (e.g., `check_dof(sb, fail_flag=True)` would raise an error when DOF!=0.)

However, observing your message here, `check_dof` may need an optional arg for a custom message and possibly some other tweaking (which is outside the scope of this PR).

Reply:
Yeah I think in this case because of the custom message its best to leave it as is. Plus at this moment I'm not sure what other functionality I'd gain by using it even if an input allowed a custom message, i.e. to check whether the degrees of freedom is 0 is only one line of code.

Reply:
Sure- my comment was more of an FYI for you and anyone else who might read through these comments. I think what you have is fine. The same applies to my comment below about the new `check_solve` function.

Reply:
I would leave it as is as well. The `check_dof` tool is rather specific to initialization use-cases where the user is one step removed from the code.